component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPUtils.java

```diff
@@ -42,35 +42,43 @@
      * @return true or false
      * @throws EmailOTPException
      */
-    public static boolean isEmailOTPEnableForLocalUser(String username, AuthenticationContext context) throws EmailOTPException {
-        String email = getEmailValueForUsername(username, context);
+    public static boolean isEmailOTPEnableForLocalUser(String username, AuthenticationContext context, boolean isUserExistence)
+            throws EmailOTPException {
+        String email = getEmailValueForUsername(username, context, isUserExistence);
         return !StringUtils.isEmpty(email);
     }
 
     /**
      * Get email value for username
+     *
      * @param username the user name
-     * @param context the authentication context
+     * @param context  the authentication context
      * @return email
      * @throws EmailOTPException
      */
-    public static String getEmailValueForUsername(String username, AuthenticationContext context) throws EmailOTPException {
+    public static String getEmailValueForUsername(String username, AuthenticationContext context, boolean isUserExistence)
+            throws EmailOTPException {
         UserRealm userRealm;
         String tenantAwareUsername = null;
         String email = null;
         try {
-            String tenantDomain = MultitenantUtils.getTenantDomain(username);
-            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-            RealmService realmService = IdentityTenantUtil.getRealmService();
-            userRealm = realmService.getTenantUserRealm(tenantId);
-            tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(String.valueOf(username));
-            if (userRealm != null) {
-                email = userRealm.getUserStoreManager()
-                        .getUserClaimValue(tenantAwareUsername, EmailOTPAuthenticatorConstants.EMAIL_CLAIM, null);
-                context.setProperty(EmailOTPAuthenticatorConstants.RECEIVER_EMAIL, email);
+            if (isUserExistence) {
```

Comment:
It does not make sense to pass a Boolean parameter and change behavior of a util method. Handle this before calling the method.